int16 robo_p_dis
int16 robo_p_drct
int16 p_exist